### PR TITLE
Adds an execution environment step to the ad hoc commands

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
This addresses some of #9864 by adding a step to select an execution environment to the ad hoc commands wizard

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

##### ADDITIONAL INFORMATION
Screen Shot 2021-04-23 at 10 48 34 AM

### CODE REVIEW

**Comment by nixocio** [comment]
Hey @AlexSCorey, 

1 - Advanced Search is not working on the EE step.

image

2 - We will need to filter EE by the Organization associated with the Inventory and the Globally Available ones.

For example

image

This group is part of demo inventory, and demo inventory is associated to Default Org.

image

EE Test should not be a choice here.

image

**Review by nixocio — CHANGES_REQUESTED** [review]
Hey @AlexSCorey, 

Just a few points that I noticed.

1 - Advanced Search is not working on the EE step.

image

2 - We will need to filter EE by the Organization associated with the Inventory and the Globally Available ones.

For example

image

This group is part of demo inventory, and demo inventory is associated to Default Org.

image

EE Test should not be a choice here.

image

File: awx/ui_next/src/components/AdHocCommands/AdHocCommands.test.jsx [hunk 1]
```diff
@@ -101,16 +135,51 @@ describe('<AdHocCommands />', () => {
   });
 
   test('should submit properly', async () => {
+    // InventoriesAPI.readAdHocOptions.mockResolvedValue({
```
Comment (nixocio) [inline awx/ui_next/src/components/AdHocCommands/AdHocCommands.test.jsx]: Remove comments.

File: awx/ui_next/src/components/AdHocCommands/AdHocCommands.jsx [hunk 2]
```diff
@@ -12,10 +12,9 @@ import AlertModal from '../AlertModal';
 import ErrorDetail from '../ErrorDetail';
 import AdHocCommandsWizard from './AdHocCommandsWizard';
 import { KebabifiedContext } from '../../contexts/Kebabified';
-import ContentLoading from '../ContentLoading';
 import ContentError from '../ContentError';
 
-function AdHocCommands({ adHocItems, i18n, hasListItems }) {
+function AdHocCommands({ adHocItems, i18n, hasListItems, onLaunchLoading }) {
```
Comment (nixocio) [inline awx/ui_next/src/components/AdHocCommands/AdHocCommands.jsx:17]: onLaunchLoading could be added as part of the `AdHocCommands.propTypes`

File: awx/ui_next/src/components/AdHocCommands/AdHocCommands.jsx [hunk 3]
```diff
@@ -36,22 +35,29 @@ function AdHocCommands({ adHocItems, i18n, hasListItems }) {
   }, [isKebabified, isWizardOpen, onKebabModalChange]);
 
   const {
-    result: { moduleOptions, credentialTypeId, isAdHocDisabled },
+    result: {
+      moduleOptions,
+      credentialTypeId,
+      isAdHocDisabled,
+      organizationId,
```
Comment (nixocio) [inline awx/ui_next/src/components/AdHocCommands/AdHocCommands.jsx:42]: 👍

File: awx/ui_next/src/components/AdHocCommands/AdHocExecutionEnvironmentStep.jsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      executionEnvironments,
+      executionEnvironmentsCount,
+      relatedSearchableKeys,
+      searchableKeys,
+    },
+  } = useRequest(
+    useCallback(async () => {
+      const params = parseQueryString(QS_CONFIG, history.location.search);
+      const globallyAvailableParams = { or__organization__isnull: 'True' };
+      const organizationIdParams = organizationId
```
Comment (nixocio) [inline awx/ui_next/src/components/AdHocCommands/AdHocExecutionEnvironmentStep.jsx:39]: 👍

**Review by nixocio — COMMENTED** [review]
Goob job @AlexSCorey.

**Comment by nixocio** [comment]
I triggered an e2e test run.

**Review by nixocio — APPROVED** [review]

**Comment by tiagodread** [comment]
@AlexSCorey I've found some items so far:

- The `Execution Environment` word only in the second step is breaking the line
- In the step 3 if you click back to the second step, we are showing `Launch` instead of `Next`
chrome-capture (7)

- Doing any action with the toolbar component in the second step like sort, search is updating the hosts list behind the modal
chrome-capture (8)

**Comment by AlexSCorey** [comment]
Tiago and I spoke offline.  I feel like the side nav issue should be addressed by PF so I filed an issue with them.  I will also have a temporary fix in place for our work

**Review by tiagodread — APPROVED** [review]

**Pull request merged**